### PR TITLE
Adjust plugin filename scan for better performance

### LINKED ISSUE #3512: HacsPluginRepository.update_repository triggers asyncio execution time warnings
### System Health details

n/a

### Checklist

- [X] I'm running the newest version of HACS <https://github.com/hacs/integration/releases/latest>
- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] I have read <https://hacs.xyz/docs/issues>
- [X] This issue is related to the backend (integration part) of HACS.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [open](https://github.com/hacs/integration/issues) or issues [pending release](https://github.com/hacs/integration/issues?q=is%3Aissue+is%3Aclosed+sort%3Aupdated-desc+milestone%3Anext).

### Describe the issue

I'm pretty sure its the time complexity of `update_filenames` triggering this

### Reproduction steps

Enable asyncio debug, restart HA

### Debug logs

```text
2024-02-15 10:31:55.990 WARNING (MainThread) [asyncio] Executing <Task pending name='Task-12346' coro=<HacsPluginRepository.update_repository() running at /config/custom_components/hacs/utils/decorator.py:28> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/local/lib/python3.11/asyncio/base_events.py:427> cb=[gather.<locals>._done_callback() at /usr/local/lib/python3.11/asyncio/tasks.py:764] created at /usr/local/lib/python3.11/asyncio/tasks.py:680> took 0.398 seconds
2024-02-15 10:31:56.207 WARNING (MainThread) [asyncio] Executing <Task pending name='Task-12347' coro=<HacsPluginRepository.update_repository() running at /config/custom_components/hacs/utils/decorator.py:28> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/local/lib/python3.11/asyncio/base_events.py:427> cb=[gather.<locals>._done_callback() at /usr/local/lib/python3.11/asyncio/tasks.py:764] created at /usr/local/lib/python3.11/asyncio/tasks.py:680> took 0.217 seconds
2024-02-15 10:31:56.480 WARNING (MainThread) [asyncio] Executing <Task pending name='Task-12344' coro=<HacsPluginRepository.update_repository() running at /config/custom_components/hacs/utils/decorator.py:28> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/local/lib/python3.11/asyncio/base_events.py:427> cb=[gather.<locals>._done_callback() at /usr/local/lib/python3.11/asyncio/tasks.py:764] created at /usr/local/lib/python3.11/asyncio/tasks.py:680> took 0.273 seconds
```
```


### Diagnostics dump

_No response_

### PR DESCRIPTION
**Pull request by ludeeus**
Fixes #3512

### CODE REVIEW

File: custom_components/hacs/repositories/plugin.py [hunk 1]
```diff
@@ -124,11 +124,13 @@ def update_filenames(self) -> None:
...
 
-        for location in ("",) if self.repository_manifest.content_in_root else ("dist", ""):
-            for filename in valid_filenames:
-                if f"{location+'/' if location else ''}{filename}" in [
-                    x.full_path for x in self.tree
-                ]:
-                    self.data.file_name = filename.split("/")[-1]
-                    self.content.path.remote = location
-                    break
+        all_paths = tuple(x.full_path for x in self.tree)
```
Comment (bdraco) [inline custom_components/hacs/repositories/plugin.py:127]:
```suggestion
        all_paths = {x.full_path for x in self.tree}
```

Would this make sense to use a `set` here so it doesn't have to do a linear search of the the tuple for the `in`/contains?

**Comment by bdraco** [comment]
Confirmed fixed with this change 👍

File: custom_components/hacs/repositories/plugin.py (hunk 1 above)
Comment (ludeeus) [reply custom_components/hacs/repositories/plugin.py:127]: Its just a collection of strings, so it does not really matter how its stored

File: custom_components/hacs/repositories/plugin.py [hunk 2]
```diff
@@ -99,9 +99,9 @@ async def get_package_content(self):
 
     def update_filenames(self) -> None:
         """Get the filename to target."""
-        # Handler for plug requirement 3
-        if self.repository_manifest.filename:
-            valid_filenames = (self.repository_manifest.filename,)
+        content_in_root = self.repository_manifest.content_in_root
+        if spesific_filename := self.repository_manifest.filename:
+            valid_filenames = (spesific_filename,)
```
Comment (bdraco) [inline custom_components/hacs/repositories/plugin.py:104]:
```suggestion
        if specific_filename := self.repository_manifest.filename:
            valid_filenames = (specific_filename,)
```

**Review by bdraco — APPROVED** [review]

File: custom_components/hacs/repositories/plugin.py (hunk 2 above)
Comment (ludeeus) [reply custom_components/hacs/repositories/plugin.py]: Thanks 🙈

**Pull request merged**